analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Loops.cs

```diff
@@ -169,18 +167,6 @@ public void Loops_For_ComplexCondition_MultipleVariables()
         var j = validator.Symbol("j");
         validator.ValidateExitReachCount(1);
         validator.TagStates("End").Should().SatisfyRespectively(
-            x =>
-            {
-                x[i].Should().HaveOnlyConstraints(ObjectConstraint.NotNull, NumberConstraint.From(10, null));
-                x[j].Should().HaveOnlyConstraints(ObjectConstraint.NotNull, NumberConstraint.From(9));
-                x[arg].Should().HaveOnlyConstraints(ObjectConstraint.NotNull, TestConstraint.First);
-            },
-            x =>
-            {
-                x[i].Should().HaveOnlyConstraints(ObjectConstraint.NotNull, NumberConstraint.From(1, 9));
-                x[j].Should().HaveOnlyConstraints(ObjectConstraint.NotNull, NumberConstraint.From(null, 0));
-                x[arg].Should().HaveOnlyConstraints(ObjectConstraint.NotNull, TestConstraint.First);
-            },
             x =>
```

Comment:
Why there are the final values that we have? For example the terminaction condition for `j=0` is
```
x[i].Should().HaveOnlyConstraints(ObjectConstraint.NotNull, NumberConstraint.From(2, 9));
x[j].Should().HaveOnlyConstraints(ObjectConstraint.NotNull, NumberConstraint.From(0));
```
I would expect `i` to be 0..9 here

Reply:
Similar situation is on line 173. Why 0-8 and not 1-10?

Reply:
When executing this code, the loop will iterate 10 times.
The SE engine will iterate it twice and determine that it cannot leave during these iterations. Afterward, it assumes that the loop will be left in a later iteration. In any case, the increment/decrement has happened at least twice.